Fix RA/TD turret facings sometimes breaking during make anim

reaperrr opened the pull request:
Fixes #18618.
Fixes default facing of Gun Turrets (compared to make anim and original, we were 1 sprite facing off).
Fixes default facing of RA Flak cannon (compared to make anim and original, we were 2 sprite facings off).
Enables recoil anim for Gun Turrets (not for AA cannon, since it looks bad with its very high fire rate).

reaperrr committed:
Fix default facing of RA/TD Gun Turrets

Compare with make anim and original, 224 was
one facing too far.

reaperrr committed:
Fix RA/TD defense turrets facing issues

By using WithSpriteTurret instead of
WithEmbeddedTurretSpriteBody.

reaperrr committed:
Enable recoil anim on RA/TD Gun Turrets

pchote commented on mods/cnc/rules/structures.yaml:
```diff
@@ -851,10 +851,15 @@ GUN:
-	-WithSpriteBody:
-	WithEmbeddedTurretSpriteBody:
+		RequiresCondition: !build-incomplete
+	WithSpriteBody:
```
`WithSpriteBody` is already inherited from `^Defense` so no need to define it again here or in RA.

pchote commented on mods/ra/rules/structures.yaml:
```diff
@@ -604,8 +604,11 @@ AGUN:
 		TurnSpeed: 60
 		InitialFacing: 896
```
This is wrong too, can we correct it to 832?

pchote reviewed (dismissed):
This is a very clever way to combine a bugfix with adding the recoil state (that I hadn't even realized we were missing!).

Just a couple of minor points:

reaperrr commented:
Updated.

abcdefg30 commented on mods/cnc/sequences/structures.yaml:
```diff
@@ -435,14 +435,15 @@ wood:
 
 gun:
-	idle:
+	idle: gunmake
```
What's going on here?

reaperrr replied:
We need an invisible placeholder for the sprite body, and the first frame of that animation is empty.

A similar approach has been used (and approved) in D2k and TS too, iirc.

abcdefg30 replied:
Can't we just null out the idle sequence of that WSB?

reaperrr replied:
Nope, WSB requires a non-null/empty sequence, and I'm not up to refactoring that (not that I'm sure we should allow WSB to work without a sequence to begin with).

abcdefg30 replied:
Ok. Can we have a comment here instead?

reaperrr replied:
Added.

reaperrr commented:
Updated:
- added comments
- fixed a small oversight, first frame of the agunmake wasn't empty (now uses gunmake frame 1, too)
- applied principle on RA SAM site as well

abcdefg30 reviewed (approved):
Ok.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/2ef77a081c3347b1b4d575543a00ee950b14e96c...5889d375ed7f973b46a4a8048755e8c04fadebbb)
